dns/delegation_set.tf

```diff
@@ -0,0 +1,17 @@
+# ------------------------------------------------------------------------------
+# Create the primary reusable delegation set.
+# It is imperative that this delgation set never be deleted.
+# ------------------------------------------------------------------------------
+
+resource "aws_route53_delegation_set" "main" {
+  # We can't perform this action until our policy is in place.
+  depends_on = [
+    aws_iam_role_policy_attachment.provisionroute53_policy_attachment,
+    aws_iam_policy.provisionroute53_policy
+  ]
+
+  lifecycle {
+    prevent_destroy = true
+  }
+  reference_name = "cool-primary"
```

Comment:
Perhaps in the interest of reusability of this code by other organizations, the reference name should be a variable?

Reply:
I could just leave the name out.  It doesn't need to be in there.  Once I saw it in the state it seemed like overkill.  I'll just call it `primary`.